StepDefinition/LanguageStepDefinitions.cs

```diff
@@ -0,0 +1,138 @@
+
+using NUnit.Framework;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using QAMars.Pages;
+using QAMars.Utilities;
+using System;
+using System.Linq;
+using System.Text;
+using TechTalk.SpecFlow;
+using System.Reflection.Emit;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Security.Cryptography.X509Certificates;
+using OpenQA.Selenium.DevTools.V128.Page;
+
+namespace ConsoleApp2.QAMars.StepDefinition
+{
+    [Binding]
+    public class LanguageStepDefinitions: CommonDriver
+    {
+        [Given(@"I logged into Mars portal successfully")]
+        public void GivenILoggedIntoMarsPortalSuccessfully()
+        {
+            
+            //login page object initiallization and definition
+            LoginPage loginPageObj = new LoginPage(driver);
+            loginPageObj.LoginActions(driver);
+        }
+        [When(@"I add a language ""([^""]*)"" and language level ""([^""]*)""")]
+        public void WhenIAddALanguageAndLanguageLevel(string language, string level)
+        {
+            //Add language page object initialization and definition
+            LanguageTest LanguagePageObj = new LanguageTest(driver);
+            LanguagePageObj.ClearData(driver);
+            LanguagePageObj.AddLanguage(driver,language, level);
+
+        }
+        //Verify if language record is created successfully
+
+        [Then(@"the ""([^""]*)"" record should be created successfully")]
+        public void ThenTheRecordShouldBeCreatedSuccessfully(string language)
+        {
+            LanguageTest LanguagePageObj = new LanguageTest(driver);
+            string newValue = LanguagePageObj.GetNewLanguage(driver,language);
+            Assert.That(newValue == language, "Actual value and expected value donot match");
```

Comment:
and you should clean up test data after but delete only the data tests adds 

see 

example before adding language you need to make sure there are no existing languages as this could result in false positive test pass
you will need to think about this for all the scenarios, this should be done before you start any tests 

the state is say that you have a test that is adding a language english, the second time you run this test you should delete any existing language so that the state is reset, this will guarantee that your test does not produce false negatives and passes always. You can not assume that test execution order is not always the same, imagine if you run the tests  in parallel mode the all tests will run at the same time so you need to also make sure you do not wipe out other tests data
hence why when you remove data that other tests are using.  The best thing to do is delete only data you added for the test to be correct
e.g test 1 added english and french as language test 2 added hindi then when you reset data you already know what language you are adding for each test
there for on reset of the two test you should delete english and french for test 1 and hindi for test 2

So to sum up you should delete all data before the start of any test and then after each test you should delete the data that a particular test is adding!
This is ensure that upon staring the test pack you have reset state and upon finishing each test deletes their data that they added!
Think about first clean up before any test runs (you delete all data around this feature e.g  before any language test run you delete all languages). Then after each test delete only data that the test has added! (test1 added english and test2 to added french then test1 should only delete english and test2 should delete french only) 

also say you are updating a language then you must added language then update it do not assume the language is there! This applies for delete or any scenario 


note you need to clean call data before any test runs and then clean data that the test generates at the end of each test! (do not clean all the data but only the one that the test is generating).

Reply:
I have implemented clear data prior to test run. Also for Edit and Delete case I have added create data in the step definition file.

Reply:
how have you done this?